x-pack/platform/plugins/private/product_intercept/public/plugin.ts

```diff
@@ -9,153 +9,196 @@ import { Subscription } from 'rxjs';
 import { PromptTelemetry } from './telemetry';
-import { TRIGGER_DEF_ID } from '../common/constants';
+import { TRIGGER_DEF_ID, UPGRADE_TRIGGER_DEF_PREFIX_ID } from '../common/constants';
 
 interface ProductInterceptPluginStartDeps {
   intercepts: InterceptsStart;
+  cloud: CloudStart;
 }
 
 export class ProductInterceptPublicPlugin implements Plugin {
   private readonly telemetry = new PromptTelemetry();
   private interceptSubscription?: Subscription;
+  private upgradeInterceptSubscription?: Subscription;
+  private readonly buildVersion: string;
+
+  constructor(ctx: PluginInitializerContext) {
+    this.buildVersion = ctx.env.packageInfo.version;
+  }
 
   setup(core: CoreSetup) {
     return this.telemetry.setup({ analytics: core.analytics });
   }
 
-  start(core: CoreStart, { intercepts }: ProductInterceptPluginStartDeps) {
+  start(core: CoreStart, { intercepts, cloud }: ProductInterceptPluginStartDeps) {
     const eventReporter = this.telemetry.start({
       analytics: core.analytics,
     });
 
-    this.interceptSubscription = intercepts
-      .registerIntercept?.({
-        id: TRIGGER_DEF_ID,
-        steps: [
-          {
-            id: 'start',
-            title: i18n.translate('productIntercept.prompter.step.start.title', {
-              defaultMessage: 'Help us improve Kibana',
-            }),
-            content: () =>
-              React.createElement(
-                EuiText,
-                { key: 'productInterceptPrompterStartContent', size: 's' },
-                i18n.translate('productIntercept.prompter.step.start.content', {
+    const productOffering = `Elastic ${function (string: string) {
+      return string.charAt(0).toUpperCase() + string.slice(1).toLowerCase();
+    }.call(null, cloud.serverless?.projectType || '')}`.trim();
+
+    void (async () => {
+      const currentUser = await core.security.authc.getCurrentUser();
+
+      const surveyUrl = new URL('https://elastic.eu.qualtrics.com/jfe/form/SV_4OfUp18saHW14Mu');
+
+      surveyUrl.searchParams.set('uid', String(currentUser.profile_uid || null));
+      surveyUrl.searchParams.set('pid', String(cloud.serverless.projectId || null));
+      surveyUrl.searchParams.set('solution', String(cloud.serverless.projectType || null));
+
+      [this.upgradeInterceptSubscription, this.interceptSubscription] = [
+        TRIGGER_DEF_ID,
+        `${UPGRADE_TRIGGER_DEF_PREFIX_ID}:${this.buildVersion}`,
+      ].map((triggerId) =>
+        intercepts
+          .registerIntercept?.({
+            id: triggerId,
+            steps: [
+              {
+                id: 'start',
+                title: i18n.translate('productIntercept.prompter.step.start.title', {
+                  defaultMessage: 'Help us improve {productOffering}',
+                  values: {
+                    productOffering,
+                  },
+                }),
+                content: () =>
+                  React.createElement(
```

Comment:
Since this whole `intercepts
          .registerIntercept`  is async and pretty large, should we move it into separate file and make it tsx to use jsx? + load into bundle as a separate chunk? Maybe even when we actually need to show the prompt

Reply:
👌🏾 ... I made some changes I'll be pushing soon.